### PR TITLE
Fix: #7362 file extension validation in create function

### PR DESCRIPTION
**Pull request by sourabpramanik**
## What does this PR do?
This PR fixes the file input upload extension validation in create function. First found in the create function step 2.
fix

## Test Plan
Manual test

## Related PRs and Issues
Fixes [#7362](https://github.com/appwrite/console/issues/853)

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?
YES

### CODE REVIEW

**Review by stnguyen90 — CHANGES_REQUESTED** [review]
Hmm, I tested this but was able to add an invalid file by dragging and dropping:

image

**Comment by sourabpramanik** [comment]
Interesting. I will look into this again and make required changes. Thanks for pointing that out ✌️

**Comment by sourabpramanik** [comment]
Hey @stnguyen90 , I see you are checking the storage file upload but this issue was not found in storage. It was found in create manual Function step 2. You can check the gif I added to the PR.

**Review by stnguyen90 — COMMENTED** [review]
@sourabpramanik here's the same in the create manual function wizard:

image

**Comment by sourabpramanik** [comment]
Hmm gotta check that again I was unable to reproduce this last time. Thanks for pointing that out.

File: src/lib/elements/forms/inputFile.svelte [hunk 1]
```diff
@@ -47,6 +36,11 @@
         hovering = false;
         if (!ev.dataTransfer.items) return;
         for (let i = 0; i < ev.dataTransfer.items.length; i++) {
+            const fileExtension = ev.dataTransfer.items[i].getAsFile().name.split('.')[1];
+            if (!allowedFileExtensions.includes(fileExtension)) {
```
Comment (stnguyen90) [inline src/lib/elements/forms/inputFile.svelte:40]: If `allowedFileExtensions` is an empty array, all file extensions should be allowed.

File: src/lib/elements/forms/inputFile.svelte [hunk 2]
```diff
@@ -81,6 +75,20 @@
 
     const handleChange = (event: Event) => {
         const target = event.currentTarget as HTMLInputElement;
+
+        const isValidFiles = Array.from(target.files).every((file) => {
+            const fileExtension = file.name.split('.').pop();
+            return allowedFileExtensions?.length
+                ? allowedFileExtensions.includes(fileExtension)
+                : true;
+        });
```
Comment (stnguyen90) [inline src/lib/elements/forms/inputFile.svelte:84]: Can the code be refactored so the logic for validating the extensions is in one function and used in both `handleChange()` and `dropHandler()`

**Review by stnguyen90 — CHANGES_REQUESTED** [review]

File: src/lib/elements/forms/inputFile.svelte (hunk 1 above)
Comment (sourabpramanik) [reply src/lib/elements/forms/inputFile.svelte:40]: Noted

File: src/lib/elements/forms/inputFile.svelte (hunk 2 above)
Comment (sourabpramanik) [reply src/lib/elements/forms/inputFile.svelte:84]: Yes sure

**Review by stnguyen90 — APPROVED** [review]

**Pull request merged**